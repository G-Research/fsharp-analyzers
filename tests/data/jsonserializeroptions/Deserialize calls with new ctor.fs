module M

open System.Text.Json

let f (json: string) (jsonStream: System.IO.Stream) =
    let _ = JsonSerializer.Deserialize<string>(json, new JsonSerializerOptions ())
    let _ = JsonSerializer.DeserializeAsync<string>(jsonStream, new JsonSerializerOptions ())
    let _ = JsonSerializer.DeserializeAsyncEnumerable<string>(jsonStream, new JsonSerializerOptions ())
    ()