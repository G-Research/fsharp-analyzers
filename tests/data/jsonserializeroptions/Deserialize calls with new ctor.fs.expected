GRA-JSONOPTS-001 | Warning | (6,12 - 6,82) | JsonSerializerOptions instances should be cached.
GRA-JSONOPTS-001 | Warning | (7,12 - 7,93) | JsonSerializerOptions instances should be cached.
GRA-JSONOPTS-001 | Warning | (8,12 - 8,103) | JsonSerializerOptions instances should be cached.
